tests/contrib/asyncio/test_tracer.py

```diff
@@ -85,3 +85,69 @@ def test_context_task_none(self):
         # but a new Context is still created making the operation safe
         ctx = self.tracer.get_call_context()
         ok_(ctx is not None)
+
+    @mark_asyncio
+    def test_exception(self):
+        async def f1():
+            with self.tracer.trace('f1'):
+                raise Exception('f1 error')
+
+        try:
```

Comment:
``self.assertRaises`` should already handle the exception (without a re-raise). You can remove the ``try-finally`` block in these tests.